lib/membrane/video_compositor/scene/object/texture.ex

```diff
@@ -34,4 +56,27 @@ defmodule Membrane.VideoCompositor.Scene.Object.Texture do
           transformations: [Transformation.definition()],
           resolution: output_resolution()
         }
+
+  @doc false
+  # Encode the texture to a Texture.RustlerFriendly in order to prepare it for
+  # the rust conversion.
+  @spec encode(t()) :: RustlerFriendly.t()
+  def encode(texture) do
+    encoded_transformations =
+      texture.transformations
+      |> Enum.map(fn transformation ->
+        case transformation do
+          %module{} -> module.encode(transformation)
+          module -> module.encode(transformation)
+        end
+      end)
```

Comment:
I would define this logic in the `Membrane.VideoCompositor.Transformation` module and call it from here since it depends on the `type` defined there

Reply:
I agree, but that makes the `Transformation` module the same as the `Layout` module, which you have taken issue with in the comment above